torchprime/experimental/torchax_models/splash_attn.py

```diff
@@ -63,14 +63,17 @@ def wrap_flash_attention(mask, query, key, value, decoder_segment_ids):
       v_layout=splash_attention_kernel.QKVLayout[global_v_layout],
     )
 
-    if mask is not None:
-      mask = splash_attention_mask.CausalMask(shape=(query.shape[2], query.shape[2]))
-
+    # FIX: If the input mask is None, create a default CausalMask.
```

Comment:
sorry, I'm not following logic here.
Code used to have check for mask creation, why are you removing that code and adding a comment to fix it?

Reply:
Previous code add mask if it is not None. seems to be a bug?